packages/core/src/components/Modal/Modal.tsx

```diff
@@ -42,9 +42,11 @@ const Component: FC<ModalProps> = memo(
                 if (modalRef.current) manager.remove(modalRef.current);
                 onCloseModal && onCloseModal();
             }, [onCloseModal, manager]),
-            handleBackgroundClick = useCallback(() => {
-                shouldCloseOnOutsideClick && handleCloseModal();
-            }, [shouldCloseOnOutsideClick, handleCloseModal]),
+            handleBackgroundClick = useCallback(
+                (event: React.MouseEvent<HTMLDivElement>) =>
```

Comment:
Minor detail, but we can get rid of the React object and import the MouseEvent from the import statement